src/test/munit/oracle/arrays-and-structs-xa-oracle-test-case.xml

```diff
@@ -0,0 +1,90 @@
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:batch="http://www.mulesoft.org/schema/mule/batch"
+      xmlns:munit="http://www.mulesoft.org/schema/mule/munit"
+      xmlns:munit-tools="http://www.mulesoft.org/schema/mule/munit-tools"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/batch http://www.mulesoft.org/schema/mule/batch/current/mule-batch.xsd
+        http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+        http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
+        http://www.mulesoft.org/schema/mule/munit-tools  http://www.mulesoft.org/schema/mule/munit-tools/current/mule-munit-tools.xsd
+		http://www.mulesoft.org/schema/mule/db http://www.mulesoft.org/schema/mule/db/current/mule-db.xsd
+		http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd">
+
+    <munit:config name="arrays-and-struct-source-oracle-test-case.xml">
+        <munit:parameterizations>
+            <munit:parameterization name="oracleDbPoolingListenerConfig">
+                <munit:parameters>
+                    <munit:parameter propertyName="config" value="oracleDbPoolingListenerConfig"/>
+                </munit:parameters>
+            </munit:parameterization>
+        </munit:parameterizations>
+    </munit:config>
+
+    <munit:before-test name="createATableForListenerTest">
+        <try>
+            <db:execute-script config-ref="${config}">
+                <db:sql><![CDATA[DROP TABLE  TEST_XA_TRANSACTION]]></db:sql>
+            </db:execute-script>
+            <error-handler>
+                <on-error-continue enableNotifications="true" logException="false">
+                </on-error-continue>
+            </error-handler>
+        </try>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[CREATE TABLE TEST_XA_TRANSACTION (id number)]]></db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql>create or replace type TEST_STRUCT as object (field1 varchar2(3), field2 number(2), field3 varchar2(60))</db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql>create or replace type TEST_STRUCT_COLLECTION as table of TEST_STRUCT</db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[
+create or replace procedure retain_first_struct (z in out TEST_STRUCT_COLLECTION)
+    IS
+begin
+    if z is not null and z.count > 1 then
+        z := TEST_STRUCT_COLLECTION(z(1));
+    end if;
+end;
+]]></db:sql>
+        </db:execute-ddl>
+    </munit:before-test>
+
+    <munit:test name="prepareArrayAndStructLocalXATransaction" ignore="#[java!org::mule::extension::db::DbMunitUtils::isTestIgnored('oracle')]">
+        <munit:execution>
+            <try transactionType="XA">
```

Comment:
Could you also add a test with other transactions in the flow that do not belong to transaction XA?